### PR TITLE
remove uses of deprecated asdf.resolver

### PR DESCRIPTION
**Pull request by braingram**
asdf.resolver will be deprecated in 2.15: https://github.com/asdf-format/asdf/pull/1362

A few uses were found in stdatamodels

1) fits_support adds a resolver for a non-existent path: https://github.com/spacetelescope/stdatamodels/blob/740477cfc3c5540233de5606e4da23241abc4266/src/stdatamodels/fits_support.py#L362-L366
2) jwst/datamodels/multiexposure provides resolvers that are equivalent to the defaults https://github.com/spacetelescope/stdatamodels/blob/740477cfc3c5540233de5606e4da23241abc4266/src/stdatamodels/jwst/datamodels/multiexposure.py#L89-L98
3) jwst/datamodels/schema_editor uses the default resolver to resolve schema references https://github.com/spacetelescope/stdatamodels/blob/740477cfc3c5540233de5606e4da23241abc4266/src/stdatamodels/jwst/datamodels/schema_editor.py#L397-L415

1 and 2 are unnecessary and this PR removes these uses.

schema_editor duplicates some of the code internal to asdf [see asdf.schema._load_schema_cached](https://github.com/asdf-format/asdf/blob/6b4701277bb25c1d4e8583b88bc2b125be9a36d6/asdf/schema.py#L498-L517). The uses of resolver appear unnecessary as the call to `generic_io.resolve_uri` and `schema.load_schema` appear sufficient to resolve the schema references (as tested with the jwst regression test: https://github.com/spacetelescope/jwst/blob/master/jwst/regtest/test_schema_editor.py).

~A complete run of the jwst regression tests with these changes will be needed.~ EDIT: jwst regression tests passed here: https://github.com/spacetelescope/stdatamodels/pull/128#issuecomment-1447086873

Resolves [JP-nnnn](https://jira.stsci.edu/browse/JP-nnnn)
Resolves [RCAL-nnnn](https://jira.stsci.edu/browse/RCAL-nnnn)

Closes #

This PR addresses ...

**Checklist**
- [ ] added entry in `CHANGES.rst` (either in `Bug Fixes` or `Changes to API`)
- [ ] updated relevant tests
- [ ] updated relevant documentation
- [ ] updated relevant milestone(s)
- [ ] added relevant label(s)

### CODE REVIEW

**Comment by braingram** [comment]
JWST regression tests run against this branch passed without errors: https://plwishmaster.stsci.edu:8081/job/RT/job/JWST/2243

JWST CI failure I believe is due to: https://github.com/spacetelescope/stdatamodels/pull/129

**Comment by braingram** [comment]
@tapastro @stscieisenhamer mentioned you may have run the schema_editor for the past few builds. Would you have time to look over the changes in this PR? Let me know if you have any questions.

**Comment by tapastro** [comment]
Sorry it took me a while to get back to you, @braingram! I don't know if there is good test coverage of the schema_editor, but I can try running it off this branch if you think this change may have affected functionality. To be honest, while I have used the code multiple times, the code you've modified looks completely unfamiliar and I'm not sure what would change on the face of it.

**Comment by braingram** [comment]
The new JWST CI failure is due to a version conflict (jwst requires stdatamodels <1.2, this repo is currently >=1.2 so the stpipe entry points fail to load with a `VersionConflict`):
https://github.com/spacetelescope/stdatamodels/actions/runs/4315086232/jobs/7528977635#step:12:848
A PR for stpipe is open that should fix this:
https://github.com/spacetelescope/stpipe/pull/84

File: src/stdatamodels/jwst/datamodels/schema_editor.py [hunk 1]
```diff
@@ -407,11 +404,10 @@ def resolve_refs(node, json_id):
                     suburl_path = suburl[:-(len(fragment) + 1)]
                 else:
                     suburl_path = suburl
-                suburl_path = resolver(suburl_path)
 
                 try:
                     subschema = aschema.load_schema(suburl_path,
-                                                    resolver,
+                                                    None,
                                                     True)
```
[the remark this answers is not in the document]
Comment (braingram) [reply src/stdatamodels/jwst/datamodels/schema_editor.py]: Good idea. I made this change.

File: src/stdatamodels/jwst/datamodels/schema_editor.py [hunk 2]
```diff
@@ -394,8 +393,6 @@ def resolve_references(self, url):
         """
         Resolve urls in the schema
         """
-        resolver = get_default_resolver()
-
         def resolve_refs(node, json_id):
```
[the remark this answers is not in the document]
Comment (braingram) [reply src/stdatamodels/jwst/datamodels/schema_editor.py:396]: It's possible this isn't needed but given the lack of test coverage and that it sounds like this is a tool that's manually run I'm less inclined to make changes unless they're absolutely necessary. Perhaps @tapastro has some thoughts on this.

**Review by WilliamJamieson — APPROVED** [review]

**Pull request merged**